### PR TITLE
PG-435: Adding new counters that can be tracked in PG15

### PR DESCRIPTION
**Pull request by codeforall**
In line with pg_stat_statments for PG15, This commit adds eight new cumulative counters for jit operations making it easier to diagnose how JIT is used in an installation. And two new columns, temp_blk_read_time, and temp_blk_write_time respectively show the time spent reading and writing temporary file blocks on disk.

### CODE REVIEW

File: pg_stat_monitor.c [hunk 1]
```diff
@@ -1671,7 +1709,7 @@ pg_stat_monitor_internal(FunctionCallInfo fcinfo,
 	if (get_call_result_type(fcinfo, NULL, &tupdesc) != TYPEFUNC_COMPOSITE)
 		elog(ERROR, "pg_stat_monitor: return type must be a row type");
 
-	if (tupdesc->natts != 52)
+	if ( tupdesc->natts != expected_out_args )
```
Comment (EngineeredVirus) [inline pg_stat_monitor.c:1712]: I think this check should be replaced by an assertion. I don't see this every happening in production.

File: pg_stat_monitor.c [hunk 2]
```diff
@@ -3343,8 +3400,8 @@ pg_stat_monitor_settings(PG_FUNCTION_ARGS)
 		}
 		else
 		{
-			values[j++] = Int64GetDatumFast(get_conf(i)->guc_min);
-			values[j++] = Int64GetDatumFast(get_conf(i)->guc_max);
+			values[j++] = Int32GetDatum(get_conf(i)->guc_min);
+			values[j++] = Int32GetDatum(get_conf(i)->guc_max);
```
Comment (EngineeredVirus) [inline pg_stat_monitor.c:3404]: Why is this change needed?

File: pg_stat_monitor.c [hunk 3]
```diff
@@ -1817,7 +1855,7 @@ pg_stat_monitor_internal(FunctionCallInfo fcinfo,
 		}
 
 		/* state at column number 8 */
-		values[i++] = Int64GetDatumFast(tmp.state);
+		values[i++] = Int64GetDatumFast((int64)tmp.state);
```
Comment (EngineeredVirus) [inline pg_stat_monitor.c:1858]: Why the explicit cast?

File: pg_stat_monitor.c (hunk 2 above)
Comment (codeforall) [reply pg_stat_monitor.c:3404]: First, it was wrong to use 64Get* macro on 32-bit variable, Second PostgreSQL code [commit](https://github.com/postgres/postgres/commit/c8b2ef05f481ef06326d7b9f3eb14b303f215c7e) adds a static assertion to GetDatum* macros to ensure type correctness, so this was always going to throw a compiler error in the future.

File: pg_stat_monitor.c (hunk 1 above)
Comment (codeforall) [reply pg_stat_monitor.c:1712]: Agreed
